src/React/Basic/Hooks.purs

```diff
@@ -84,20 +71,29 @@ component ::
   Lacks "ref" props =>
   String ->
   ({ | props } -> Render Unit hooks JSX) ->
-  CreateComponent { | props }
+  Effect (ReactComponent { | props })
 component name renderFn =
   let
-    c = unsafeReactFunctionComponent (mkEffectFn1 (\props -> case renderFn props of Render a -> a))
+    c =
+      unsafeReactFunctionComponent
+        ( mkEffectFn1
+            ( \props ->
+                unsafeDiscardRenderEffects (renderFn props)
+            )
+        )
   in
     runEffectFn2 unsafeSetDisplayName name c
 
+unsafeDiscardRenderEffects :: forall x y a. Render x y a -> Effect a
+unsafeDiscardRenderEffects = unsafeCoerce
```

Comment:
Opted to keep this an `unsafeCoerce` instead of exporting another unsafe function from Internal

Reply:
Also AFAIK `unsafeCoerce`s are erased at compile-time.

Reply:
Oh neat

Reply:
Well, probably at `purs bundle` time, but we use webpack :)